test/integration/ads_integration_test.cc

```diff
@@ -1749,149 +1749,172 @@ TEST_P(AdsIntegrationTest, ContextParameterUpdate) {
+                    route_name_1);
   sendDiscoveryResponse<envoy::config::listener::v3::Listener>(
-      Config::TypeUrl::get().Listener, {},
-      {buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
-                     "baz?xds.node.cluster=cluster_name&xds.node.id=node_name",
-                     route_name_1)},
-      {}, "2");
+      Config::TypeUrl::get().Listener, {second_listener}, {second_listener}, {}, "2");
+
   EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "1", {},
-                                      {route_name_1}, {}));
+                                      {route_name_1}, {}, /*expect_node=*/isSotw()));
+  const auto second_route_config = buildRouteConfig(route_name_1, cluster_name);
   sendDiscoveryResponse<envoy::config::route::v3::RouteConfiguration>(
-      Config::TypeUrl::get().RouteConfiguration, {}, {buildRouteConfig(route_name_1, cluster_name)},
-      {}, "2");
+      Config::TypeUrl::get().RouteConfiguration, {second_route_config}, {second_route_config}, {},
+      "2");
 
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "2", {}, {}, {}));
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "2", {}, {}, {}));
-  test_server_->waitForCounterEq("listener_manager.listener_create_success", 2);
-  makeSingleRequest();
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "2", {}, {}, {},
+                                      /*expect_node=*/isSotw()));
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "2", {}, {}, {},
+                                      /*expect_node=*/isSotw()));
 
-  // Update bar listener in the foo namespace.
-  sendDiscoveryResponse<envoy::config::listener::v3::Listener>(
-      Config::TypeUrl::get().Listener, {},
-      {buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
-                     "bar?xds.node.cluster=cluster_name&xds.node.id=node_name",
-                     route_name_1)},
-      {}, "3");
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "3", {}, {}, {}));
-  test_server_->waitForCounterEq("listener_manager.listener_in_place_updated", 1);
+  test_server_->waitForCounterEq("listener_manager.listener_create_success", 2);
   makeSingleRequest();
 
-  // Remove bar listener from the foo namespace.
-  sendDiscoveryResponse<envoy::config::listener::v3::Listener>(
-      Config::TypeUrl::get().Listener, {}, {},
-      {"xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
-       "bar?xds.node.cluster=cluster_name&xds.node.id=node_name"},
-      "3");
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "4", {}, {}, {}));
-  test_server_->waitForCounterEq("listener_manager.listener_removed", 1);
-  makeSingleRequest();
+  // Updates and deletions only apply to the Delta protocol, not SotW.
```

Comment:
Not sure I understand this... for LDS (wildcard) removal can be done by sending the rest of the listeners.